python/graphstorm/gpartition/metis_partition.py

```diff
@@ -0,0 +1,201 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Parmetis partition assignment
+"""
+import os
+import logging
+import json
+import subprocess
+
+from .partition_algo_base import LocalPartitionAlgorithm
+
+
+class ParMetisPartitionAlgorithm(LocalPartitionAlgorithm):
+    """
+    Multiple-instances metis partitioning algorithm.
+
+    The partition algorithm accepts the intermediate output from GraphStorm
+    gs-processing which matches the requirements of the DGL distributed
+    partitioning pipeline.
+
+
+    Parameters
+    ----------
+    metadata_dict: dict
+        DGL "Chunked graph data" JSON, as defined in
+        https://docs.dgl.ai/guide/distributed-preprocessing.html#specification
+    metis_config: ParMETISConfig
+        Configuration object for ParMETIS.
+    """
+
+    def __init__(self, metadata_dict, metis_config):
+        super().__init__(metadata_dict)
+        self.metis_config = metis_config
+
+    def _launch_preprocess(self, num_parts, input_path, ip_list, dgl_tool_path, metadata_filename):
+        """ Launch preprocessing script
+
+        Parameters
+        ----------
+        num_parts: int
+            Number of graph partitions
+        input_path: str
+            Path to the input graph data
+        ip_list: str
+            ip list file after port assigned
+        dgl_tool_path: str
+            Path to the dgl tool added in the PYTHONPATH
+        metadata_filename: str
+            Meta data configuration name
+        """
+        command = f"mpirun -np {num_parts} --allow-run-as-root --hostfile {ip_list} \
+                    -wdir {input_path} \
+                  python3 {dgl_tool_path}/distpartitioning/parmetis_preprocess.py \
+                    --input_dir {input_path} \
+                    --schema_file {metadata_filename} \
+                    --output_dir {input_path} --num_parts {num_parts}"
+
+        if self.run_command(command):
+            logging.info("Successfully execute parmetis preprocess.")
+            return True
+        else:
+            logging.info("Failed to execute parmetis preprocess.")
+            return False
+
+
+    def _launch_parmetis(self, num_parts, input_path, ip_list, graph_name):
+        """ Launch parmetis script
+
+        Parameters
+        ----------
+        num_parts: int
+            Number of graph partitions
+        input_path: str
+            Path to the input graph data
+        ip_list: str
+            ip list
+        graph_name: str
+            Graph name
+        """
+        command = f"mpirun -np 1 --allow-run-as-root \
+                    --hostfile {ip_list} \
+                    --mca orte_base_help_aggregate 0 -mca btl_tcp_if_include eth0 \
+                    -wdir {input_path} \
+                ~/local/bin/pm_dglpart {graph_name} {num_parts} {input_path}/parmetis_nfiles.txt \
```

Comment:
Actually it is more like hardcoded in the PM4GNN installation guide.(Enterprise version of parmetis). After ```make install ```, it will install the dependencies there. I think for our users, once they are using the docker image, which we would highly recommend after this PR, they should not care the details like that. What we would better to do here is to keep everything on the same page to avoid any problem.

Reply:
Another idea I just come up with is that I can add an assert here to avoid any mismatch problem to make sure the pm_dglpart is in the right place. But not to provide users to specify the pm_dglpart path themselves.